### PR TITLE
Remove tcp mock from tests

### LINKED ISSUE #2337: Remove tcp mock from tests
shame, but needed to be compatible with the Rust bindings

### PR DESCRIPTION
**Pull request by touilleMan**
fix #2337

### CODE REVIEW

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -213,7 +213,7 @@ jobs:
     displayName: 'Install'
   - bash: |
       set -eux
-      py.test $(pytest.base_args) tests -n auto --runrust
+      python tests/run_parallel.py -n auto $(pytest.base_args) tests --runrust
```
Comment (vxgmichel) [inline .azure-pipelines.yml:216]: Why is `-n auto` still here if `run_parallel.py` already runs the tests in parallel?

Comment (vxgmichel) [reply .azure-pipelines.yml:216]: Nevermind, I get it now.

**Review by vxgmichel — APPROVED** [review]
We should be good to go

**Comment by touilleMan** [comment]
image

**Pull request merged**